tests/grading/test_export_bonuspoints.py

```diff
@@ -0,0 +1,192 @@
+        suppressed during bonus points calculation.
+        """
+        stdout, file_contents, path = self.export_bonuspoints(
+            self.task.category.name, datetime(1970, 1, 1)
+        )
+
+        self.assertIn("Successfully created {}".format(path), stdout)
+        self.assertEqual(len(file_contents), 0)
+
+        merged_file_contents = "".join(file_contents)
+
+        self.assertNotIn("alice", merged_file_contents)
+        self.assertNotIn("bob", merged_file_contents)
+
+
+class PlagiarismInfluencedVetoCounteredBonusPointsExportTest(
+    DetectedPlagiarismVetoData, BonusPointsTestCase
+):
+    def test_export_with_active_veto(self):
+        """
+        Validate that solutions with detected plagiarisms are
+        not suppressed if they contain an active veto.
+        """
+        stdout, file_contents, path = self.export_bonuspoints(
+            self.task.category.name, datetime(1970, 1, 1)
+        )
+
+        self.assertIn("Successfully created {}".format(path), stdout)
+        self.assertEqual(len(file_contents), 2)
+
+        merged_file_contents = "".join(file_contents)
+
+        self.assertIn(",,alice,alice@example.org,,,1", merged_file_contents)
+        self.assertIn(",,bob,bob@example.org,,,1", merged_file_contents)
+
+
+class BonusPointsExportTimingTest(SimpleAccountsData, SimpleTaskData, BonusPointsTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.passed_solution_1 = Solution.objects.create(
+            author=cls.alice,
+            task=cls.task,
+            passed=True,
+        )
+        cls.passed_solution_2 = Solution.objects.create(
+            author=cls.alice,
+            task=cls.task,
+            passed=True,
+        )
+
+    def test_export_bonus_points_most_recent_solution(self):
+        """
+        Validate that only the most recent passed solution
+        is taken into account for bonus points calculation.
+        """
+        stdout, file_contents, path = self.export_bonuspoints(
+            self.task.category.name, datetime.now()
+        )
+
+        self.assertIn("Successfully created {}".format(path), stdout)
+        self.assertEqual(len(file_contents), 1)
+
+        merged_file_contents = "".join(file_contents)
+
+        self.assertIn("alice", merged_file_contents)
+
+    def test_export_bonus_points_future_start_date(self):
+        """
+        Validate that start_date is taken into account for
+        bonus points calculation.
+        """
+        stdout, file_contents, path = self.export_bonuspoints(
+            self.task.category.name, datetime.now() + timedelta(days=1337)
+        )
+
+        self.assertIn("Successfully created {}".format(path), stdout)
+        self.assertEqual(len(file_contents), 0)
+
+        merged_file_contents = "".join(file_contents)
+
+        self.assertNotIn("alice", merged_file_contents)
+
+
+@tag("slow")
+class BonusPointsExportPlagiarismTimingTest(
```

Comment:
This test case is still too complex and fragile and needs some more work.

Reply:
For now, I removed the test, because it was too complex and fragile. I will think about a better solution and integrate that one instead.